app/gui/src/controller/searcher/component/builder.rs

```diff
@@ -164,9 +168,14 @@ impl List {
         }
     }
 
-    /// Build the list, sorting all group lists and groups' contents appropriately. (Does not sort
-    /// the [`component::List::favorites`].)
-    pub fn build(self) -> component::List {
+    /// Build the list, sorting all group lists and groups' contents appropriately. Filter the
+    /// [`component::List::favorites`] (only components with IDs passed to [`extend`] are
+    /// retained), do not sort them.
+    ///
+    /// If a component group in favorites is empty after the filtering, the empty group is
+    /// retained. This allows layoing out the favorites in [Component
+    /// Browser](crate::controller::Searcher) in the same columns regardless of filtering.
```

Comment:
The information that only those favorites will be kept which have IDs among those passed to extend shall be added to the `set_favorites` method. 

And I miss a description _why_ the builder behaves this way. There should be a "usage scenario" in the docs of the `builder::List` struct: that we create a list, set favorites _structure_, extend with IDs matching return/self type and get list with properly filtered groups.